aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -362,33 +426,74 @@ sub pxeprint
     $fh->print (<<EOF
 # File generated by pxelinux AII plug-in.
 # Do not edit.
-default $t->{label}
-    label $t->{label}
-    kernel $t->{kernel}
+default Install $pxe_config->{label}
+    label $pxe_config->{label}
+    kernel $pxe_config->{kernel}
     $appendtxt
 EOF
            );
     # TODO is ksdevice still mandatory? if not, fix schema (code is already ok)
     # ksdecvice=bootif is an anaconda-ism, but can serve general purpose
-    $fh->print ("    ipappend 2\n") if ($t->{ksdevice} && $t->{ksdevice} eq 'bootif');
+    $fh->print ("    ipappend 2\n") if ($pxe_config->{ksdevice} && $pxe_config->{ksdevice} eq 'bootif');
     $fh->close();
 }
 
-# Prints an IP address in hexadecimal.
-sub hexip
+
+# Write the Grub2 configuration file.
+# Return 1 if the file was written successfully, 0 otherwise.
+# TODO: handle append options?
```

Comment:
I opened https://github.com/quattor/aii/issues/255 to list all the miscellanous things to do that are not strictly required for the first version...